### DISCUSSION TITLE
Can `deepMap`'s `setByKey` function be exported?

### DISCUSSION THREAD

**Discussion by yhdgms1** [ai/nanostores#295] [General]
Right now only the `setPath` that under the hood uses the `setByKey` function is exported. However, `setPath` accepts only a string path, when `setByKey` accepts an array path.

I wrote a wrapper around deepMap, and I want to update store's value like it done in `setKey` method. But I can't because it requires string and not an array which I only have.

I can copy that function and functions it depends on or mock used methods like below
```ts
setPath($store, { split: () => ({ flatMap: () => myPathToValue }) }, value)
```
But using function directly would be better.

So I think `setByKey` function can be exported.

**Comment by ai** [comment]
Sure, send PR.
